src/app/state/staking/saga.ts

```diff
@@ -93,17 +93,50 @@ function* refreshValidators() {
         address: v.account_id,
         name: v.account_name,
         escrow: v.escrow_balance,
-        fee: v.fee,
+        fee: 0, // set below
         status: v.status,
         media: v.media_info,
         rank: index + 1,
       } as Validator
     })
 
+  for (let i = 0; i < validators.length; i++) {
+    payload[i].fee = (yield* call(getValidatorFee, payload[i].address)) ?? 0
+  }
+
   yield* put(stakingActions.updateValidators(payload))
   return validators
 }
 
+function* getValidatorFee(address: string) {
+  const nic = yield* call(getOasisNic)
+  const publicKey = yield* call(addressToPublicKey, address)
+  const account = yield* call([nic, nic.stakingAccount], { owner: publicKey, height: 0 })
+  const currentEpoch = yield* select(selectEpoch) || 0
+
+  let rawRates = account.escrow?.commission_schedule?.rates
+  if (!rawRates) {
+    rawRates = []
+  }
+
+  const rates: Rates[] = rawRates
+    .map(r => ({
+      epochStart: r.start ? Number(r.start) : 0,
+      rate: r.rate ? Number(quantity.toBigInt(r.rate)) / 100_000 : undefined,
+    }))
+    .sort((a, b) => a.epochStart - b.epochStart)
```

Comment:
https://github.com/oasisprotocol/oasis-wallet-web/blob/d28f7438/src/app/pages/StakingPage/Features/CommissionBounds/index.tsx#L58 should also sort just in case then

Reply:
maybe

Reply:
This is copy&paste from the getValidatorDetails() function below https://github.com/oasisprotocol/oasis-wallet-web/pull/298/files#diff-3462f431342207098b94164c722136313b5d8d7456385f74bf9c89720dd89ebdR157